src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -543,107 +536,107 @@ public Mono<RedirectUrlResponseDto> requestRedirectUrlAuthorization(
-                )
-                .idPaymentMethod(idPaymentMethod.toString())
-                .paName(null);// optional
-        Either<CheckoutRedirectConfigurationException, URI> pspConfiguredUrl = getRedirectUrlForPsp(
-                authorizationData.pspId()
-        );
-        return pspConfiguredUrl.fold(
-                Mono::error,
-                proxyPspUrl -> nodeForwarderRedirectApiClient
-                        .proxyRequest(
-                                request,
-                                proxyPspUrl,
-                                authorizationData.transactionId().value(),
-                                RedirectUrlResponseDto.class
-                        ).onErrorMap(
-                                NodeForwarderClientException.class,
-                                exception -> {
-                                    String pspId = authorizationData.pspId();
-                                    if (exception
-                                            .getCause()instanceof WebClientResponseException webClientResponseException) {
-
-                                        HttpStatus httpStatus = webClientResponseException.getStatusCode();
-                                        log.error(
-                                                "Error communicating with PSP: [{}] to retrieve redirection URL. Received HTTP status code: {}",
-                                                pspId,
-                                                httpStatus
-                                        );
-                                        return switch (httpStatus) {
-                                            case BAD_REQUEST, UNAUTHORIZED -> new AlreadyProcessedException(
-                                                    authorizationData.transactionId()
-                                            );
-                                            default -> new BadGatewayException(
-                                                    "KO performing redirection URL api call for PSP: [%s]"
-                                                            .formatted(pspId),
-                                                    httpStatus
-                                            );
-                                        };
-                                    } else {
-                                        return new BadGatewayException(
-                                                "Unhandled error performing redirection URL api call for PSP: [%s]"
-                                                        .formatted(pspId),
-                                                null
-                                        );
-                                    }
-                                }
+        return new JwtTokenUtils()
+                .generateToken(
+                        npgNotificationSigningKey,
+                        npgJwtKeyValidityTime,
```

Comment:
```suggestion
                        npgNotificationSigningKey,
                        npgJwtKeyValidityTime,
```
This token is the one that will be used by eCommerce f.e. for polling the transaction status in the outcome page. Please use the eCommerce signing key here

Reply:
Excellent observation fixed with 56689e716478674f1d3421ef9eb15b8ebe407ecc